onedal/datatypes/data_conversion.cpp

```diff
@@ -155,18 +155,20 @@ dal::table convert_to_table(PyObject *obj) {
     }
     if (is_array(obj)) {
         PyArrayObject *ary = reinterpret_cast<PyArrayObject *>(obj);
-        if (array_is_behaved_C(ary) || array_is_behaved_F(ary)) {
+        if (!PyArray_ISCARRAY_RO(ary) && !PyArray_ISFARRAY_RO(ary)) {
+            // NOTE: this will make a C-contiguous deep copy of the data
+            // this is expected to be a special case
+            ary = PyArray_GETCONTIGUOUS(ary);
```

Comment:
The Decref is necessary to prevent a memory leak (otherwise leaks in BasicStatitistics occur in all CIs), the memory is effectively owned by the table object, who will call its destructor with the pycapsule.  This work makes to_table support non-contiguous arrays, which will simplify the python coding occurring before to_table.  This work will enable ```_check_array``` to be moved from ```onedal``` to ```sklearnex``` simplifying the rollout of the new finite checker to SVM and neighbors algorithms. There is no error handling available (that I can find) in the numpy C-api as also numpy generally doesn't raise errors like that.  The checks that it is 1) a numpy array and 2) of certain aligned and type characteristics will prevent errors, and the use of the numpy c-api is very standardized.

Reply:
It looks like it can fail to allocate, in which case it will return `NULL`:
https://github.com/numpy/numpy/blob/cf9598572528318a54489b3c9ed5f65ef042e8c8/numpy/_core/src/multiarray/convert.c#L495

Reply:
Ahhh thank you for doing that research @david-cortes-intel , that would mean in the case of a failure in allocation, it will fail to decref. I will add a check there.